alicloud/service_alicloud_ots.go

```diff
@@ -0,0 +1,45 @@
+package alicloud
+
+import (
+	"github.com/aliyun/aliyun-tablestore-go-sdk/tablestore"
+)
+
+func getPrimaryKeyType(primaryKeyType string) tablestore.PrimaryKeyType {
+	var keyType tablestore.PrimaryKeyType
+	t := PrimaryKeyType(primaryKeyType)
+	switch t {
+	case IntegerType:
+		keyType = tablestore.PrimaryKeyType_INTEGER
+	case StringType:
+		keyType = tablestore.PrimaryKeyType_STRING
+	case BinaryType:
+		keyType = tablestore.PrimaryKeyType_BINARY
+	}
+	return keyType
+}
+
+func describeOtsTable(tableName string, meta interface{}) (*tablestore.DescribeTableResponse, error) {
+	client := meta.(*AliyunClient).otsconn
+
+	describeTableReq := new(tablestore.DescribeTableRequest)
+	describeTableReq.TableName = tableName
+
+	describ, err := client.DescribeTable(describeTableReq)
+	return describ, err
+}
+
+func deleteOtsTable(tableName string, meta interface{}) (bool, error) {
+	client := meta.(*AliyunClient).otsconn
+
+	deleteReq := new(tablestore.DeleteTableRequest)
+	deleteReq.TableName = tableName
+	_, err := client.DeleteTable(deleteReq)
+
+	describ, _ := describeOtsTable(tableName, meta)
+
```

Comment:
If describ is nil, the next code will be throw a nil pointer error. So I suggest method describeOtsTable does not return a pointer.

Reply:
In ots table describetable api https://github.com/aliyun/aliyun-tablestore-go-sdk/blob/master/tablestore/api.go?#L347:L389, here are two lines for returning.

return &DescribeTableResponse{}, err
return response, nil in this situation, response is an object of DescribeTableResponse response := new(DescribeTableResponse)
So describe could not be a nil, but its properties like TableMeta could not nil which can be used to check whether the returned ots table object is nil or not. Thanks:)